modules/tinymce/src/plugins/link/main/ts/core/Utils.ts

```diff
@@ -81,13 +81,14 @@ const hasLinks = (elements: Node[]) => Tools.grep(elements, isLink).length > 0;
 
 const hasLinksInSelection = (rng: Range) => collectNodesInRange(rng, isLink).length > 0;
 
-const isOnlyTextSelected = (html: string) => {
-  // Partial html and not a fully selected anchor element
-  if (/</.test(html) && (!/^<a [^>]+>[^<]+<\/a>$/.test(html) || html.indexOf('href=') === -1)) {
-    return false;
-  }
-
-  return true;
+const isOnlyTextSelected = (editor: Editor) => {
+  // Allow anchor and inline text elements to be in the selection but nothing else
+  const inlineTextElements = editor.schema.getTextInlineElements();
+  const isElement = (elm: Node): elm is Element => elm.nodeType === 1 && !isAnchor(elm) && !Obj.has(inlineTextElements, elm.nodeName.toLowerCase());
```

Comment:
Why's the type matter in the helper function?

Reply:
Maybe instead of 1 use `NodeTypes.ELEMENT` from sugar?

Reply:
We try to keep plugins small, so this compiles to a smaller value. Also this is a type guard, it's required for the rest of the logic.

Reply:
I should also add you'll see that we use the "magic" numbers all throughout the existing code for node type. We maybe should reconsider that one day, but that's a discussion for outside a PR.